### PR TITLE
yash: init at 2.52

### PR DESCRIPTION
**Pull request by qbit**
###### Motivation for this change

yash is a small, feature rich, POSIX compliant shell. It offers a lot of the features of a shell like zsh, but without a lot of the complexity.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by buckley310** [comment]
LGTM.
Built and used it for a couple minutes on nixos 21.11.

File: pkgs/shells/yash/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+stdenv.mkDerivation rec {
+  pname = "yash";
+  version = "2.52";
+
+  src = fetchurl {
+    url = "https://osdn.net/dl/yash/yash-${version}.tar.xz";
+    sha256 = "sha256:1jdmj4cyzwxxyyqf20y1zi578h7md860ryffp02qi143zpppn4sm";
+  };
+
+  meta = with lib; {
```
Comment (AndersonTorres) [inline pkgs/shells/yash/default.nix:12]: `meta` should be the next-to-latest attribute of an expression.

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -13589,4 +13589,15 @@
     github = "PhilippWoelfel";
     githubId = 19400064;
   };
+  qbit = {
```
Comment (AndersonTorres) [inline maintainers/maintainer-list.nix:13592]: Welcome aboard!

The info about you should be put in a separated commit. Please, split accordingly.

File: pkgs/shells/yash/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    description =
+      "Yet another shell, is a POSIX-compliant command line shell written in C99 (ISO/IEC 9899:1999)";
```
Comment (AndersonTorres) [inline pkgs/shells/yash/default.nix:14]: A more concise description:

```suggestion
    description = "Yet another POSIX-compliant shell";
```

File: pkgs/shells/yash/default.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    homepage = "https://yash.osdn.jp/index.html.en";
+    license = licenses.gpl2;
```
Comment (AndersonTorres) [inline pkgs/shells/yash/default.nix:16]:
```suggestion
    license = licenses.gpl2Plus;
```

File: pkgs/shells/yash/default.nix [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+  src = fetchurl {
+    url = "https://osdn.net/dl/yash/yash-${version}.tar.xz";
+    sha256 = "sha256:1jdmj4cyzwxxyyqf20y1zi578h7md860ryffp02qi143zpppn4sm";
+  };
+
+  meta = with lib; {
+    description =
+      "Yet another shell, is a POSIX-compliant command line shell written in C99 (ISO/IEC 9899:1999)";
+    homepage = "https://yash.osdn.jp/index.html.en";
```
Comment (AndersonTorres) [inline pkgs/shells/yash/default.nix:15]: Optionally, put homepage first.

File: pkgs/shells/yash/default.nix [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    license = licenses.gpl2;
+    platforms = platforms.all;
+    maintainers = maintainers.qbit;
```
Comment (AndersonTorres) [inline pkgs/shells/yash/default.nix:18]: Alphabetical ordering is easier to read.

```suggestion
    maintainers = maintainers.qbit;
    platforms = platforms.all;
```

File: pkgs/shells/yash/default.nix [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  };
+
+  buildInputs = [ ncurses ];
+
+  propagatedBuildInputs = [ gettext ];
```
Comment (AndersonTorres) [inline pkgs/shells/yash/default.nix:23]: Move this next to `src` attribute.

```suggestion
  buildInputs = [ 
    gettext
    ncurses
  ];
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by AndersonTorres** [comment]
ALSO, remember to put your commit on the format 

`yash: init at <version>`

File: pkgs/shells/yash/default.nix [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+  passthru = { shellPath = "/bin/yash"; };
```
Comment (AndersonTorres) [inline pkgs/shells/yash/default.nix:25]:
```suggestion
  passthru.shellPath = "/bin/yash";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/shells/yash/default.nix (hunk 6 above)
Comment (pennae) [reply pkgs/shells/yash/default.nix:18]: `maintainers` should also be a list: `with maintainers; [ qbit ]`

File: maintainers/maintainer-list.nix (hunk 2 above)
Comment (qbit) [reply maintainers/maintainer-list.nix:13592]: Thanks! fixed!

File: pkgs/shells/yash/default.nix (hunk 5 above)
Comment (qbit) [reply pkgs/shells/yash/default.nix:16]: I opted for sticking with alphabetical

**Comment by qbit** [comment]
ok, I think I fixed all the things brought up!

**Comment by AndersonTorres** [comment]
LGTM, LGTBorg.

**Pull request merged**